### PR TITLE
Introduce a st,stm32g0-exti compatible

### PR DESCRIPTION
**Pull request by Desvauxm-st**
Introduce a st,stm32g0-exti compatible
 added to the matching targets:
C0/G0/U5/L5/MP:

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Miss binding entry for new compatible

Would be nice to get the first commit title more accurate:
"drivers: gpio/exti: stm32: Use st,stm32g0-exti compatible"

File: dts/bindings/interrupt-controller/st,stm32g0-exti.yaml
Comment (erwango) [inline dts/bindings/interrupt-controller/st,stm32g0-exti.yaml:1]: We don't need to provide a full description. st,stm32g0-exti is used for this.
But we need to explain the goal behind the introduction of this new binding, see st,stm32h7-spi.yaml as example

Comment (Desvauxm-st) [reply dts/bindings/interrupt-controller/st,stm32g0-exti.yaml:1]: ok

File: dts/bindings/interrupt-controller/st,stm32g0-exti.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  interrupt-names:
+    required: true
+
+  num-lines:
+    type: int
+    required: true
+
+  line-ranges:
+    type: array
+    required: true
```
Comment (erwango) [inline dts/bindings/interrupt-controller/st,stm32g0-exti.yaml:30]: Remove all these as they are already provided with st,stm32-exti

**Review by erwango — DISMISSED** [review]

File: dts/bindings/interrupt-controller/st,stm32g0-exti.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+        STM32 controller
+        This compatible stands for all interrupt-controller blocks for the
+        STM32C0/G0/L5/mp/u5.
+
+compatible: "st,stm32g0-exti"
+
+include: [base.yaml, interrupt-controller.yaml]
```
Comment (gautierg-st) [inline dts/bindings/interrupt-controller/st,stm32g0-exti.yaml:12]: Shouldn't we include [st,stm32-exti.yaml] instead, like it is done for example for uart/usart, to remove the following compilation error:
devicetree error: 'num-lines' appears in /soc/interrupt-controller@46022000 in /__w/zephyr/zephyr/twister-out/b_u585i_iot02a/tests/drivers/clock_control/stm32_clock_configuration/stm32u5_core/drivers.stm32_clock_configuration.u5.sysclksrc_pll_hsi_40/zephyr/zephyr.dts.pre, but is not declared in 'properties:' in /__w/zephyr/zephyr/dts/bindings/interrupt-controller/st,stm32g0-exti.yaml

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Also, the two commits should be switched, the first one should introduce the new compatible, and the second should replace the SOC defines.

File: dts/bindings/interrupt-controller/st,stm32g0-exti.yaml (hunk 2 above)
Comment (Desvauxm-st) [reply dts/bindings/interrupt-controller/st,stm32g0-exti.yaml:12]: OK

**Review by erwango — APPROVED** [review]

File: drivers/gpio/gpio_stm32.c [hunk 3]
```diff
@@ -297,12 +297,7 @@ static inline uint32_t gpio_stm32_pin_to_exti_line(int pin)
...
 	defined(CONFIG_SOC_SERIES_STM32F0X)
 	return ((pin % 4 * 4) << 16) | (pin / 4);
-#elif defined(CONFIG_SOC_SERIES_STM32MP1X)
-	return (((pin * 8) % 32) << 16) | (pin / 4);
-#elif defined(CONFIG_SOC_SERIES_STM32C0X) || \
-	defined(CONFIG_SOC_SERIES_STM32G0X) || \
-	defined(CONFIG_SOC_SERIES_STM32L5X) || \
-	defined(CONFIG_SOC_SERIES_STM32U5X)
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32g0_exti)
 	return ((pin & 0x3) << (16 + 3)) | (pin >> 2);
```
Comment (FRASTM) [inline drivers/gpio/gpio_stm32.c:301]: Is the returned value also valid for stm32MP1 ?
(previously `return (((pin * 8) % 32) << 16) | (pin / 4);`)

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**